### PR TITLE
Some more minor lexer refactoring

### PR DESCRIPTION
**Pull request by milseman**

### CODE REVIEW

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
@hamishknight @natecook1000 I went ahead and merged to make other diffs cleaner, but I'm still interested in reviews.

File: Sources/Util/MissingUnicode.swift [hunk 1]
```diff
@@ -0,0 +1,983 @@
+// MARK: - Missing stdlib API
+
+extension Unicode {
+  public enum Script: String {
```
Comment (hamishknight) [inline Sources/Util/MissingUnicode.swift:4]: We may want to separate out the in-source spellings as UTS#18 has fuzzy matching rules and aliases for properties (e.g ignoring case, whitespace, and `_`, and e.g `Latn` being an alias for `Latin`), but that can be done when we implement the lexing/parsing for them.

**Review by hamishknight — COMMENTED** [review]
LGTM

File: Sources/Util/MissingUnicode.swift (hunk 1 above)
Comment (milseman) [reply Sources/Util/MissingUnicode.swift:4]: CC @natecook1000, resident expert on character classes and properties.

**Pull request merged**